src/rdkafka_cgrp.c

```diff
@@ -3762,15 +3762,11 @@ rd_kafka_consumer_group_metadata_new (const char *group_id) {
 }
 
 rd_kafka_consumer_group_metadata_t *
-rd_kafka_consumer_group_metadata_new2 (const char *group_id,
-                                       int32_t generation_id) {
+rd_kafka_consumer_group_metadata_new_with_genid (const char *group_id,
+                                                 int32_t generation_id) {
         rd_kafka_consumer_group_metadata_t *cgmetadata;
 
-        if (!group_id)
-                return NULL;
-
-        cgmetadata = rd_calloc(1, sizeof(*cgmetadata));
-        cgmetadata->group_id = rd_strdup(group_id);
+        cgmetadata = rd_kafka_consumer_group_metadata_new(group_id);
```

Comment:
nit: I would flip this the other way, have metadata_new() call the richer new_with_genid(), so that there is only a single real constructor.

Reply:
whoops. i was struggling with a 35 degree day in an oven with no aircon. round 2 today.

Reply:
+1 for C!